authenticator/types.go

```diff
@@ -35,9 +35,27 @@ type TokenResponse struct {
 // when making a call to the Authenticate method
 // It returns the status of the response, headers and the response body
 type AuthenticationResponse struct {
-	Status  int         `json:"status"`
-	Headers http.Header `json:"headers"`
-	Body    interface{} `json:"body,omitempty"`
+	Status  int
+	Headers http.Header
+	body    interface{}
+}
+
+// Error returns the ErrorBody of the authentication response
+func (a *AuthenticationResponse) Error() *ErrorBody {
+	if a.Status != http.StatusOK {
+		return a.body.(*ErrorBody)
+	}
+
+	return nil
+}
+
+// TokenResponse returns the token returned by the response
+func (a *AuthenticationResponse) TokenResponse() *TokenResponse {
+	if a.Status != http.StatusOK {
```

Comment:
Copy-paste error?

Reply:
Why? the return value is `nil` if we have a non success status code.

Reply:
I can't read.